gazebo/rendering/Heightmap.cc

```diff
@@ -523,7 +523,9 @@ void Heightmap::Load()
   {
     this->dataPtr->splitTerrain = true;
     nTerrains = this->dataPtr->numTerrainSubdivisions;
-    prefix = terrainDirPath / "gazebo_terrain_cache";
+    std::string terrainName = "gazebo_terrain_cache" +
+        std::to_string(this->LOD());
```

Comment:
I've noticed that the cache files appear to be identical for any `LOD > 0`. So I don't think we need to use separate file names for each `LOD` value, just one for `LOD == 0` and one for `LOD > 0`.

How about the following?

~~~
std::string terrainNameSuffix;
if (this->LOD() == 0)
{
  terrainNameSuffix = "_LOD0";
}
std::string terrainName = "gazebo_terrain_cache" + terrainNameSuffix;
~~~

also, to avoid code duplication `terrainNameSuffix` could also be computed before this `if` branch so it could also be used in the `else` branch